sdk/samples/router/general.py

```diff
@@ -0,0 +1,320 @@
+        ),
+        endpoint="/echo",
+        timeout="60ms",
+        port=8080,
+        env=[],
+    )
+
+    # Create the RouterConfig instance
+    router_config = RouterConfig(
+        environment_name="id-dev",
+        name="what-does-the-fox-say",
+        routes=routes,
+        rules=rules,
+        default_route_id="test",
+        experiment_engine=experiment_config,
+        resource_request=resource_request,
+        timeout="100ms",
+        log_config=log_config,
+        enricher=enricher,
+        ensembler=ensembler
+    )
+
+    # 1. Create a new router using the RouterConfig object
+    new_router = turing.Router.create(router_config)
+    print(f"1. You have created a router with id: {new_router.id}")
+
+    # 2. List all routers
+    routers = turing.Router.list()
+    print(f"2. You have just retrieved a list of {len(routers)} routers:")
+    for r in routers:
+        if r.name == new_router.name:
+            my_router = r
+        print(r)
+
+    # Wait for the router to get deployed
+    try:
+        my_router.wait_for_status(RouterStatus.DEPLOYED)
+    except TimeoutError:
+        raise Exception(f"Turing API is taking too long for router {my_router.id} to get deployed.")
+
+    # 3. Get the router you just created using the router_id obtained
+    my_router = turing.Router.get(my_router.id)
+    print(f"3. You have retrieved the router with name: {my_router.name}")
+
+    # Access the router config from the returned Router object directly
+    my_router_config = my_router.config
+
+    # Modify something in the router config
+    my_router_config.routes.append(
+        Route(
+            id='fee-fi-fo-fum',
+            endpoint='http://fox-says.fee-fi-fo-fum',
+            timeout='20ms'
+        )
+    )
+
+    # 4. Update the router with the new router config
+    my_router.update(my_router_config)
+    print(f"4. You have just updated your router with a new config.")
+
+    # 5. List all the router config versions of your router
+    my_router_versions = my_router.list_versions()
+    print(f"5. You have just retrieved a list of {len(my_router_versions)} versions for your router:")
+    for ver in my_router_versions:
+        print(ver)
+
+    # Sort the versions returned by version number
+    my_router_versions.sort(key=lambda x: x.version)
+    # Get the version number of the first version returned
+    first_ver_no = my_router_versions[0].version
+    # Get the version number of the latest version returned
+    latest_ver_no = my_router_versions[-1].version
+
+    # Wait for the latest version to get deployed
+    try:
+        my_router.wait_for_version_status(RouterStatus.DEPLOYED, latest_ver_no)
+    except TimeoutError:
+        raise Exception(f"Turing API is taking too long for router {my_router.id} with version {latest_ver_no} to get "
+                        f"deployed.")
+
+    # Wait for the dependencies of the first version to be fully undeployed
+    time.sleep(15)
```

Comment:
What is it for? Can it be done as: ?
```python
my_router.wait_for_version_status(RouterStatus.UNDEPLOYED, first_ver_no)
```

Reply:
Actually this is also similar to the issue right below; I was kind of trying to test both 'potential sources' of error (either with something undeploying being undeployed completely/properly, or with something being deployed not being deployed completely/properly) in both places.